pandas/core/apply.py

```diff
@@ -530,7 +530,8 @@ def series_generator(self):
                 ser._mgr = mgr
                 blk.values = arr
                 ser.name = name
-                yield ser
+                # To fix issue of #39166
+                yield ser.copy()
```

Comment:
the entire point of series_generator is to avoid creating a bunch of Series objects.  if this is too buggy to be viable, might as well remove the  whole method and just use `self.obj.iloc[i] for i in range(len(self.obj))`